### DISCUSSION TITLE
Hardening ABS

### DISCUSSION THREAD

**Discussion by darthrater78** [advplyr/audiobookshelf#3273] [General]
So I've exposed my ABS install to the internet via cloudflare DNS proxy that points to another NPM proxy in my DMZ to get to the server on my production Network. I've enabled prevent common exploits on npm, and set geoblocking on my inbound rule.

Are there any other common sense tactics I can employ to harden this further in terms of the server itself?

**Comment by jaycedk** [comment]
Fail2ban to block ip addresses, if they get through you geoblocking, and try to brute force there way in.

**Comment by jeeftor** [comment]
There are 2 VERY simple options:

1) Cloud Flare Tunnels - if you already have Cloudflare DNS then these are for free - look in the Zero Trust section
2) Tailscale ...

**Reply by deddc23efb** [reply]
How does cloudflare tunnels add any additional protection?  Does the Cloudflare proxy do anything more than what NPM would do?
Tailscale only works if you don't want to share with non-tech friends or family.

**Comment by jeeftor** [comment]
Tunnels handle the auth for you and only forward a specific port (ish) of your choice.
